remote-trait-object-tests/src/test_ping.rs

```diff
@@ -0,0 +1,42 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use super::mod_main::main_like as main_main;
+use super::mod_ping::main_like as ping_main;
+use crate::connection::{create_connection, ConnectionEnd};
+
+/// There are thee entities: commander, main module and ping module.
+/// Commander sends "start" message to the main module.
+/// If the main module receives "start" message, it sends "ping" to the ping module.
+/// If the ping module receives "ping" message, respond "pong".
+/// If the main module received "pong" response, send "pong received" to the commander.
+#[test]
+fn ping() {
+    let (main_to_cmd, cmd_to_main) = create_connection();
+    let (main_to_ping, ping_to_main) = create_connection();
+
+    main_main(Vec::new(), main_to_cmd, main_to_ping);
+    ping_main(Vec::new(), ping_to_main);
+
+    let ConnectionEnd {
+        sender: to_main,
+        receiver: from_main,
+    } = cmd_to_main;
+
+    to_main.send("request:start".to_string()).unwrap();
```

Comment:
Would you consider making this ping-pong repetitive

Reply:
I think the current once "ping-pong" is enough to show the usage of the module.
What do you hope by making the example repetitive?
When you suggest a thing, always tell why it is better.

Reply:
There might be a bug when we try to send/recv multiple packets, or it might be turned out to be really slow.
If you think this simple case is important, then how about adding an additional test that is repetitive?
It is always better to have an additional harsh test.

Reply:
I don't think the harsh test is required in this case.
We know the result code and the performance will be checked in the result code.
Maintaining a harsh test now makes writing code hard.

What I want to do now is migrating your code into small pieces. I do not agree with running a harsh test. Also running a harsh test should be implemented after we implement the interface. 
The interface is more important than performance. We can make code more performant easily with a good interface. However, changing the performant code to have a better interface is very difficult.

Reply:
Ok then.